golem-worker-executor-base/src/services/rdbms/sqlx_common.rs

```diff
@@ -712,7 +712,7 @@ where
         let mut tx_conn = self.tx_connection.0.lock().await;
 
         if tx_conn.open {
-            DB::TransactionManager::start_rollback(&mut tx_conn.connection);
+            let _ = DB::TransactionManager::rollback(&mut tx_conn.connection).await;
```

Comment:
This was the reason for the flakiness in the sql tests I reported a few days ago.

What happens here is the rollback is started asynchronously on the connection, but it is getting returned to the pool.
What can happen then is that somebody else picks up the connection from the pool and starts running work on it. The rollback is then interleaved with the new work, causing various weird failures.

Waiting for the the rollback to complete when dropping the connection should fix this.

@justcoon am I missing something with this one?

Reply:
nice catch